ts/components/CiePinpad/index.tsx

```diff
@@ -0,0 +1,175 @@
+import { Text, View } from "native-base";
+import * as React from "react";
+import {
+  Dimensions,
+  StyleSheet,
+  TextInput,
+  TextInputKeyPressEventData
+} from "react-native";
+
+import variables from "../../theme/variables";
+import { Baseline } from "../Pinpad/Placeholders";
+
+type Props = {
+  description: string;
+  pinLength: number;
+  onPinChanged: (pin: string) => void;
+};
+
+type State = {
+  pin: ReadonlyArray<string>;
+};
+
+const styles = StyleSheet.create({
+  placeholderContainer: {
+    flexDirection: "row",
+    justifyContent: "center",
+    backgroundColor: variables.colorWhite
+  },
+  text: {
+    alignSelf: "center",
+    justifyContent: "center",
+    color: variables.colorWhite
+  },
+  placeHolderStyle: {
+    height: 4,
+    marginLeft: 2,
+    marginRight: 2,
+    marginTop: 2
+  }
+});
+
+const width = 36;
+const margin = 2;
+const screenWidth = Dimensions.get("window").width;
+/**
+ * A customized CodeInput component.
+ */
+class CiePinpad extends React.PureComponent<Props, State> {
+  private inputs: ReadonlyArray<TextInput>;
+
+  constructor(props: Props) {
+    super(props);
+    this.inputBoxGenerator = this.inputBoxGenerator.bind(this);
+    this.inputs = [];
+    this.state = {
+      pin: new Array(this.props.pinLength).fill("")
+    };
+  }
+
+  private handleOnChangeText = (text: string, index: number) => {
+    const pin = this.updatePin(text, index);
+    this.props.onPinChanged(pin.join(""));
+    // if it is not the last element, change focus on next element
+    // handleOnKeyPress is used to handle the Backspace press
+    if (index + 1 < this.inputs.length) {
+      this.inputs[index + 1].focus();
+      return;
+    }
+  };
+
+  // tslint:disable-next-line: readonly-array
+  private updatePin = (text: string, index: number): string[] => {
+    // tslint:disable-next-line: readonly-array
+    const tempPin = [...this.state.pin];
+    // replace the pin char at the index position
+    tempPin.splice(index, 1, text);
+    const pin: ReadonlyArray<string> = tempPin;
+    this.setState({ pin });
+    return tempPin;
+  };
+
+  private handleOnKeyPress = (
+    nativeEvent: TextInputKeyPressEventData,
+    index: number
+  ) => {
+    if (nativeEvent.key === "Backspace") {
+      // check if a deletion is going. If yes, set the focus to the previous input
+      // it works only if the index is not the first element
+      if (
+        (!this.state.pin[index] || this.state.pin[index].length === 0) &&
+        index > 0
+      ) {
+        this.updatePin("", index - 1);
+        this.inputs[index - 1].focus();
+      }
+      return;
+    }
+  };
+
+  private readonly inputBoxGenerator = (i: number) => {
+    const totalMargins = margin * 2 * (this.props.pinLength - 1);
```

Comment:
```suggestion
    const sideMargin = 8;
    const totalMargins =
      margin * 2 * (this.props.pinLength - 1) + sideMargin * 2;
```
and place **_sideMargin_** on top of file.
This ensures the inputs (first and last) are not too close to the screen edges
Like in this screen (iPhone 5S)

<img src="https://user-images.githubusercontent.com/822471/68848410-93bfb480-06d0-11ea-984e-aa9e6b9f3963.png" height="600"/>

Reply:
Ok